core/src/main/scala/org/apache/spark/TaskContextImpl.scala

```diff
@@ -82,6 +83,13 @@ private[spark] class TaskContextImpl(
   // If defined, the corresponding task has been killed and this option contains the reason.
   @volatile private var reasonIfKilled: Option[String] = None
 
+  // The pending interruption request, which is blocked by uninterruptible resource creation.
+  // Should be protected by `TaskContext.synchronized`.
+  private var pendingInterruptRequest: Option[(Option[Thread], String)] = None
+
+  // Whether this task is able to be interrupted. Should be protected by `TaskContext.synchronized`.
```

Comment:
or maybe mark as `@transient`?

Reply:
Are you referring to `@volatile`?

Reply:
Since we already have `TaskContext.synchronized` protected, `@volatile` could be redundant. And if we use `@volatile` only, it's not thread safety as we could have two threads modify `_interruptible` concurrently.